src/main/java/com/fasterxml/jackson/databind/deser/std/FactoryBasedEnumDeserializer.java

```diff
@@ -132,18 +132,21 @@ public Object deserialize(JsonParser p, DeserializationContext ctxt) throws IOEx
             // 30-Mar-2020, tatu: For properties-based one, MUST get JSON Object (before
             //   2.11, was just assuming match)
             if (_creatorProps != null) {
-                if (!p.isExpectedStartObjectToken()) {
+                if (p.isExpectedStartObjectToken()) {
+                    if (_propCreator == null) {
```

Comment:
Ok I know this is existing code but this is wrong (not sure how I added such code :) ), wrt multi-threaded access.
Could you extract this into separate, synchronized method? At least it'd be slightly less bad.

Reply:
Yes.

Reply:
Oh. I missed the part that it is `transient` already. Hmmh. I guess it'd be slightly better to use `AtomicReference` but access is guarded.

Reply:
Ok, so it's `transient` but that won't do anything for access. I think what'd make sense is using `AtomicReference` here.